platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_9.adoc

```diff
@@ -42,3 +42,8 @@ The previous step will help you identify potential problems by issuing deprecati
 === Deprecations
 
 === Potential breaking changes
+
+==== Removed incubating method `ObjectFactory#dependencyCollector()`
+
+The incubating `ObjectFactory#dependencyCollector()` method has been removed.
+DependencyCollectors can still be instantiated within Gradle managed objects.
```

Comment:
💅 Maybe add a link to the documentation chapter (https://docs.gradle.org/current/userguide/properties_providers.html#managed_types) here?